documentation/developer_setup.md

```diff
@@ -1,88 +1,74 @@
+An SDK that uses code generated clients directly exposes the Autorest-generated code to consumers of the module and is the preferred approach.
+The [azkeys][azkeys_directory] module is an example of a code generated client (CGC).
 
-This client takes three parameters, the first is the service URL for the specific account. The second is an [`interface`][go_interfaces] which is a specific struct that has definitions for a certain set of methods. In the case of `azcore.TokenCredential` the `GetToken(context.Context, options policy.TokenRequestOptions)` method must be defined to be a valid interface. The final argument to methods that create clients or interact with the service should be a pointer to an `Options` parameter. This options struct should have `azcore.ClientOptions` embedded and any service specific options. Making this final parameter a pointer allows the customer to pass in `nil` if there are no specific options they want to change.
+Note that for data-plane CGCs, client constructors must be hand-written as there's no consisten form of authentication across data-plane services.
 
+### Internally Generated Clients
 
-### Defining Methods
+Internally generated clients are used when the Autorest-generated code isn't fit for direct, public consumption.
+In this design, the generated code is placed under an `/internal` directory, prohibiting it from being directly imported by module consumers,
+and all publicly exposed content is either hand-written or a type alias of internal types (see `Alias declarations` in the [Go language specification][type_declarations] for more info on creating type aliases).
 
-Defining a method follows the format:
-```golang
-// Create creates the table with the tableName specified when NewClient was called.
-func (t *Client) Create(ctx context.Context, options *CreateTableOptions) (CreateTableResponse, error) {
-	if options == nil {
-		options = &CreateTableOptions{}
-	}
-	resp, err := t.client.Create(ctx, generated.Enum1Three0, generated.TableProperties{TableName: &t.name}, options.toGenerated(), &generated.QueryOptions{})
-	return createTableResponseFromGen(&resp), err
-}
+The [aztables][aztables_directory] modules is an example that uses internally generated clients.
 
-```
-The `(s *Client)` portion is the "receiver". Methods can be defined for either pointer (with a `*`) or receiver (without a `*`) types. Pointer receivers will not copy types on method calls and allows the method to mutate the receiving struct. Client methods should use a pointer receiver.
+## Create Module Skeleton
+
+There are several files required to be in the root directory of your module.
+
+- CHANGELOG.md for tracking released changes
+- LICENSE.txt is the MIT license file
+- NOTICE.txt for legal attributions
+- README.md for getting started
+- ci.yml for PR and release pipelines
+- go.mod defines the Go module
+
```

Comment:
also need a version.go file

Reply:
Should we also include that the autorest.md and build.go files would be here?

Reply:
Added a section detailing module version constant.